packages/opc-menu-drawer/src/opc-menu-drawer.ts

```diff
@@ -15,31 +18,53 @@ export class OpcMenuDrawer extends LitElement {
 
   @property({ type: Array, reflect: true })
   links: OpcMenuDrawerLinkGroup[] = [];
-  @property({ type: String, reflect: true }) headerTitle = '';
+  @property({ type: String, reflect: true }) title = 'Menu';
+  @property({ type: String, reflect: true }) menuTitle = '';
+
   @state() private _isOpen = false;
+  @state() private _searchLinkState: Record<string, string> = {};
+  @state() private _searchDebounce: NodeJS.Timeout;
 
-  /**
-   * This is to monitor all states of collapsed boxes
-   * Why, when client height can be used?
-   * Because it doesn't sync at first, the request update is fired before client height changes
-   * So to avoid unsynced collapse state.
-   */
-  @state() private collapsableState: boolean[] = [];
+  @query('.opc-menu-drawer-menu')
+  collapsableMenuContainer: HTMLDivElement | undefined;
 
-  @queryAll('.opc-menu-drawer-menu-btn,.opc-menu-drawer-links-collapse-box')
-  collapsableLinks: NodeListOf<HTMLDivElement> | undefined;
+  private _handleLinkExpandToggle(event: Event) {
+    const container = (event.target as HTMLDivElement).parentElement!;
+    const collapseBox = container.querySelector(
+      '.opc-menu-drawer-links-collapse-box'
+    )! as HTMLDivElement;
+    const collapseIndicator = container.querySelector('.angle-icon')!;
+    this._toggleContainerHeight(collapseBox, collapseIndicator);
+  }
 
-  private _handleExpandToggle(pos: number) {
-    const toggleRef = this.collapsableLinks[pos]!;
-    this.collapsableState[pos] = !this.collapsableState[pos];
+  private _handleMenuExpandToggle() {
+    const collapsableMenu = this.collapsableMenuContainer.querySelector(
+      '.opc-menu-drawer-menu-btn'
+    ) as HTMLDivElement;
+    const collapseIndicator = this.collapsableMenuContainer.querySelector(
+      '.angle-icon'
+    );
+    this._toggleContainerHeight(collapsableMenu, collapseIndicator);
+  }
 
-    if (toggleRef.clientHeight === 0) {
-      toggleRef.style.height = toggleRef.scrollHeight + 'px';
+  private _toggleContainerHeight(box: HTMLDivElement, indicator: Element) {
+    if (box.clientHeight === 0) {
+      box.style.height = box.scrollHeight + 'px';
+      indicator.setAttribute('inverse', 'true');
     } else {
-      toggleRef.style.height = '0';
+      box.style.height = '0';
+      indicator.removeAttribute('inverse');
     }
+  }
 
-    this.requestUpdate();
+  private _handleSearchChange(type: string, search: string) {
+    clearTimeout(this._searchDebounce);
+    this._searchDebounce = setTimeout(() => {
+      this._searchLinkState = {
+        ...this._searchLinkState,
+        [type]: search.toLowerCase(),
+      };
+    }, 300);
```

Comment:
@akhilmhdh can you add a comment for this function? I'm not sure how 300ms would be helpful. Can `Promise` help? Or there's always `@change=${e => console.log(e.target.value)}` Sorry i'm not able to understand this cc @gisumit can you help.

Reply:
I applied a debounce there to control the change wrt to ui minimal. When the list is really long we dont want to re-render the list for every keystroke. Here its not a timeout delay operation its debounce operation. So what i did here is basically wait for user to finish typing by 300ms then re-render